filters/auth/grantflowstate.go

```diff
@@ -0,0 +1,132 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/zalando/skipper/secrets"
+)
+
+const randomStringLength = 20
+
+type state struct {
+	Rand       string `json:"rand"`
+	Validity   int64  `json:"validity"`
+	Nonce      string `json:"nonce"`
+	RequestURL string `json:"redirectUrl"`
+}
+
+type flowState struct {
+	secrets     *secrets.Registry
+	secretsFile string
+}
+
+var errExpiredAuthState = errors.New("expired auth state")
+
+const (
+	secretSize    = 20
+	letterBytes   = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
+)
+
+var (
+	src = rand.NewSource(time.Now().UnixNano())
+)
+
+// https://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-golang
+func randString(n int) string {
+	b := make([]byte, n)
+	// A src.Int63() generates 63 random bits, enough for letterIdxMax characters!
+	for i, cache, remain := n-1, src.Int63(), letterIdxMax; i >= 0; {
+		if remain == 0 {
+			cache, remain = src.Int63(), letterIdxMax
+		}
+		if idx := int(cache & letterIdxMask); idx < len(letterBytes) {
+			b[i] = letterBytes[idx]
+			i--
+		}
+		cache >>= letterIdxBits
+		remain--
+	}
+
+	return string(b)
+}
+
+func newFlowState(secrets *secrets.Registry, secretsFile string) *flowState {
+	return &flowState{
+		secrets:     secrets,
+		secretsFile: secretsFile,
+	}
+}
+
+func stateValidityTime() int64 {
+	return time.Now().Add(time.Hour).Unix()
+}
+
+func (s *flowState) createState(redirectURL string) (string, error) {
+	encrypter, err := s.secrets.GetEncrypter(secretsRefreshInternal, s.secretsFile)
+	if err != nil {
+		return "", err
+	}
+
+	nonce, err := encrypter.CreateNonce()
+	if err != nil {
+		return "", err
+	}
+
+	state := state{
+		Rand:       randString(randomStringLength),
```

Comment:
What is this Rand for? The state is encrypted and has a random Nonce every time it is created...

Reply:
This code was removed. As we discussed, it's not necessary.